### PR TITLE
Updating tests to force .pyi files to be committed into the repo

### LINKED ISSUE #766: [FEA]: Avoid running stubgen in CI
### Is this a new feature, an improvement, or a change to existing functionality?

Change

### How would you describe the priority of this feature request

High

### Please provide a clear description of problem this feature solves

Running stubgen in CI requires the Nvidia drivers to be installed in the CI image. This unfortunately locks us to a specific driver version, and ops would like to accelerate adopting new versions.

### Describe your ideal solution

Include the stubs as part of the source code and add a CI check to make sure they are up to date
* This would shift generating the stubs to the developer (can still be part of the build. The dev just needs to commit them)
* The test in CI could happen in the test stage which already has a GPU
* The test would be as simple as running the stubgen code and seeing if there are any diffs. Similar to what we do for clang-format

### Describe any alternatives you have considered

1) See if the cuda-toolkit driver stubs work the same as cudatoolkit
If we can load the stubs but not run anything, that would work for us. The current stubs throw an error on load.
This is a long shot but requires the least amount of work

2) Generate the stubs in a pre-commit hook

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

### PR DESCRIPTION
**Pull request by mdemoret-nv**
## Description
This PR changes the stub generation to be committed into the repo instead of as part of the build. This will allow CI steps that need the PYI files to not require a GPU as well.

This works by always copying the PYI files back to the source tree so they will show up as diffs. Any diffs during the test phase will be reported as an error.

Closes #766

### CODE REVIEW

File: morpheus.code-workspace [hunk 1]
```diff
@@ -1,623 +1,618 @@
...
-					"${workspaceFolder}": {
-						"editorPath": "${workspaceFolder}",
-						"useForBreakpoints": "true"
-					}
-				},
-			},
-		],
-		"compounds": []
-	}
+    "folders": [
```
Comment (cwharris) [inline morpheus.code-workspace:2]: Are all of these changes intentional?

**Review by cwharris — APPROVED** [review]
have you verified that the tests can fail due to mismatched `.pyi` files?

File: ci/release/update-version.sh [hunk 2]
```diff
@@ -67,7 +67,11 @@ function sed_runner() {
 # .gitmodules
 git submodule set-branch -b branch-${NEXT_SHORT_TAG} external/morpheus-visualizations
 git submodule set-branch -b branch-${NEXT_SHORT_TAG} morpheus_utils
-git submodule update --remote --recursive
+
+if [[ "$(git diff --name-only | grep .gitmodules)" != "" ]]; then
```
[the remark this answers is not in the document]
Comment (mdemoret-nv) [reply ci/release/update-version.sh:71]: No, but calling `git submodule update` will checkout the branch specified in .gitmodules resulting in a diff unless you are using the branch head.

[the remark this answers is not in the document]
Comment (cwharris) [reply ci/release/update-version.sh:71]: Might be useful to have a comment explaining as much.

**Review by dagardner-nv — APPROVED** [review]
LGTM

File: ci/release/update-version.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdemoret-nv) [reply ci/release/update-version.sh:71]: Added a comment

File: morpheus.code-workspace (hunk 1 above)
Comment (mdemoret-nv) [reply morpheus.code-workspace:2]: Yes, just sorted the JSON to make diffs easier in the future.

**Pull request merged**